package.json

```diff
@@ -100,6 +100,7 @@
     "jest-cli": "16.0.2",
     "jest-webpack-alias": "3.3.0",
     "jison-loader": "1.0.0",
+    "js-beautify": "1.6.14",
```

Comment:
Hey @tnguyen0 thanks for the changes regarding codemirror, but I am still not a fan of `js-beautify`. You are just using this for HTML formation. I would like us to omit a package if we only need a simple function from it, because one package more means one package more we need to care for and update. Even if the update is not affecting (read improving) the way we use it. 

I kind of went forward and created a short example for the indentation of HTML:
https://jsfiddle.net/onq7ooyj/6/

Also the Javascript code could easily be refactored I just grabbed it from stackoverflow from this question and added the feature that the first level will not have a newline character before it:
https://stackoverflow.com/questions/26360414/javascript-how-to-correct-indentation-in-html-string